pkg/cmd/openshift-kube-apiserver/kubeadmission/register.go

```diff
@@ -0,0 +1,136 @@
+	securityadmission.Register(plugins)
+	securityadmission.RegisterSCCExecRestrictions(plugins)
+	externalipranger.RegisterExternalIP(plugins)
+	restrictedendpoints.RegisterRestrictedEndpoints(plugins)
+}
+
+var (
+
+	// these are admission plugins that cannot be applied until after the kubeapiserver starts.
+	// TODO if nothing comes to mind in 3.10, kill this
+	SkipRunLevelZeroPlugins = sets.NewString()
+	// these are admission plugins that cannot be applied until after the openshiftapiserver apiserver starts.
+	SkipRunLevelOnePlugins = sets.NewString(
+		"ProjectRequestLimit",
+		"openshift.io/RestrictSubjectBindings",
+		"openshift.io/ClusterResourceQuota",
+		"openshift.io/ImagePolicy",
+		overrideapi.PluginName,
+		"OriginPodNodeEnvironment",
+		"RunOnceDuration",
+		sccadmission.PluginName,
+		"SCCExecRestrictions",
+	)
+
+	// BeforeKubeAdmissionPlugins is the list of plugins to add before kube admission, so we run before limit ranges
+	// TODO this cannot be done on today's mutation
+	beforeKubeAdmissionPlugins = []string{
+		"ClusterResourceOverride",
+	}
+
+	// AfterKubeAdmissionPlugins are the admission plugins to add after kube admission, before mutating webhooks
+	afterKubeAdmissionPlugins = []string{
+		"openshift.io/RestrictSubjectBindings",
+		"RunOnceDuration",
+		"PodNodeConstraints",
+		"OriginPodNodeEnvironment",
+		externalipranger.ExternalIPPluginName,
+		restrictedendpoints.RestrictedEndpointsPluginName,
+		"openshift.io/ImagePolicy",
+		sccadmission.PluginName,
+		"SCCExecRestrictions",
+		ingressadmission.IngressAdmission,
+	}
+
+	// FinalKubeAdmissionPlugins are the admission plugins to add after quota.  You shouldn't ever add this.
+	finalKubeAdmissionPlugins = []string{
+		"openshift.io/ClusterResourceQuota",
+	}
+
+	// additionalDefaultOnPlugins is a list of plugins we turn on by default that core kube does not.
+	additionalDefaultOnPlugins = sets.NewString(
+		imageadmission.PluginName, // "openshift.io/ImageLimitRange"
+		"OriginPodNodeEnvironment",
+		"PodNodeSelector",
+		"Priority",
+		externalipranger.ExternalIPPluginName,
+		restrictedendpoints.RestrictedEndpointsPluginName,
+		noderestriction.PluginName,
+		securityadmission.PluginName,
+		"StorageObjectInUseProtection",
+		"SCCExecRestrictions",
+		"PersistentVolumeLabel",
+		"OwnerReferencesPermissionEnforcement",
+		"PodTolerationRestriction",
+		"openshift.io/ClusterResourceQuota",
+		"openshift.io/IngressAdmission",
+	)
+
+	// additionalDefaultOffPlugins are admission plugins we choose not to enable by default in openshift
+	// you shouldn't put anything from kube in this list without api-approvers signing off on it.
+	additionalDefaultOffPlugins = sets.NewString(
+		"ProjectRequestLimit",
+		"RunOnceDuration",
+		"PodNodeConstraints",
+		overrideapi.PluginName,
+		imagepolicyapi.PluginName,
+		"openshift.io/RestrictSubjectBindings",
+	)
+)
+
+func NewOrderedKubeAdmissionPlugins(kubeAdmissionOrder []string) []string {
```

Comment:
Shouldn't we have a test verifying that our methods are in-sync with upstream list of admissions? This applies for both for `NewOrderedKubeAdmissionPlugins` and `NewDefaultOffPluginsFunc`

Reply:
> Shouldn't we have a test verifying that our methods are in-sync with upstream list of admissions? This applies for both for `NewOrderedKubeAdmissionPlugins` and `NewDefaultOffPluginsFunc`

I don't think so.  I the idea of this pull is that we always use the upstream order and only inject in a couple spots.  I think I'm even going to change those spots to have a single injection point in a followup pull.  If you couldn't be a webhook, you can't be included.